src/lib/registration/ui/base_system_registration_dialog.rb

```diff
@@ -78,23 +178,123 @@ def content
           VStretch(),
           product_details_widgets,
           VSpacing(Yast::UI.TextMode ? 1 : 2),
-          input_widgets,
-          VSpacing(Yast::UI.TextMode ? 0 : 1),
-          # button label
-          PushButton(Id(:local_server), _("&Local Registration Server...")),
+          registration_widgets,
           VSpacing(Yast::UI.TextMode ? 0 : 3),
-          skip_button,
           reregister_extensions_button,
           VStretch()
         )
       end
 
-      # part of the main dialog definition - display the "skip" skip button only
-      # when the system is not registered yet
-      # @return [Yast::Term] UI term
-      def skip_button
-        # button label
-        Registration.is_registered? ? Empty() : PushButton(Id(:skip), _("&Skip Registration"))
+      def registration_widgets
+        HSquash(
+          VBox(
+            RadioButtonGroup(
+              Id(:action),
+              VBox(
+                register_scc_option,
+                register_local_option,
+                skip_option
+              )
+            )
+          )
+        )
+      end
+
+      # Return registration options
+      #
+      # Read registration options from Storage::InstallationOptions
+      # and, if needed, from Storage::RegCodes.
+      #
+      # @retun [Hash] Hash containing values for :reg_code,
+      #               :email and :custom_url.
+      def reg_options
+        return @reg_options unless @reg_options.nil?
+        options = Storage::InstallationOptions.instance
+
+        reg_code = options.reg_code
+        if reg_code.empty?
+          known_reg_codes = Storage::RegCodes.instance.reg_codes
+          base_product_name = SwMgmt.find_base_product["name"]
+          reg_code = known_reg_codes[base_product_name] || ""
+        end
+
+        @reg_options = {
+          reg_code:   reg_code,
+          email:      options.email,
+          custom_url: options.custom_url || default_url
+        }
+      end
+
+      # Default registration server
+      #
+      # @return [String] URL for the registration server
+      def default_url
+        @default_url ||= SUSE::Connect::Config.new.url
+      end
+
+      def register_scc_option
+        VBox(
+          Left(
+            RadioButton(
+              Id(:register_scc),
+              Opt(:notify),
+              # TRANSLATORS: radio button
+              _("Register System via SCC.SUSE.COM"),
```

Comment:
I'd prefer lowercase SCC host name, the URL or host names are usually (maybe rather always) written in lower case.